backend/review/import_utils/import_to_db.py

```diff
@@ -0,0 +1,312 @@
+    return name
+
+
+User = get_user_model()
+
+
+def filter_by_term(rows, semesters, semester_key="TERM"):
+    return [row for row in rows if row[semester_key] in semesters]
+
+
+def import_instructor(pennid, fullname, stat):
+    """
+    Given instructor info do one of a few things:
+    1. Create a new instructor and user object and link the two
+    2. Link together any existing instructors/users that should be linked together assuming the
+       given pennid and fullname belong to the same person.
+    """
+    if pennid is not None:
+        stat("instructors_with_pennid")
+
+        user, user_created = User.objects.get_or_create(
+            pk=int(pennid), defaults={"username": uuid.uuid4()}
+        )
+        if user_created:
+            stat("users_created")
+            user.set_unusable_password()
+            user.save()
+
+        if Instructor.objects.filter(user=user).exists():
+            inst = Instructor.objects.get(user=user)
+            inst_created = False
+        elif Instructor.objects.filter(name=fullname).exists():
+            inst = Instructor.objects.get(name=fullname)
+            inst.user = user
+            inst.save()
+            inst_created = False
+        else:
+            inst = Instructor.objects.create(user=user, name=fullname)
+            inst_created = True
+
+        if inst_created:
+            stat("instructors_created")
+
+    elif len(fullname) > 0:
+        inst, inst_created = Instructor.objects.get_or_create(name=fullname)
+        if inst_created:
+            stat("instructors_created")
+    else:
+        stat("instructors_without_info")
+        inst = None
+
+    return inst
+
+
+def import_course_and_section(full_course_code, semester, course_title, primary_section_code, stat):
+    """
+    Given course and section info, update/create objects.
+    """
+    try:
+        (course, section, course_created, section_created) = get_or_create_course_and_section(
+            full_course_code, semester, report_created=True
+        )
+    except ValueError:
+        stat("invalid_section_id")
+        return None, None
+
+    # Update course title if one isn't already set.
+    course_dirty = False
+    if course.title == "":
+        course_dirty = True
+        course.title = course_title
+
+    # If no primary listing is set, add one.
+    if primary_section_code is not None and course.primary_listing is None:
+        try:
+            dept, ccode, _ = separate_course_code(primary_section_code)
+        except ValueError:
+            stat("invalid_primary_section_id")
+            return
+
+        course.primary_listing = get_or_create_course(dept, ccode, semester)
+        course_dirty = True
+
+    if course_dirty:
+        stat("courses_updated")
+        course.save()
+
+    return course, section
+
+
+def import_review(section, instructor, enrollment, responses, form_type, bits, stat):
+    # Assumption: that all review objects for the semesters in question were
```

Comment:
Does this apply to historical data or just the current semester?

Reply:
There is not `PCR_RATING_HIST` file... all rating details from 2002 onwards are in this one gb-large file 🙃

Reply:
No difference between the two -- see the main Command class where we delete all reviews which have a matching semester field before importing.

Reply:
Is that good behavior though? I feel like it would make more sense to just import semesters you specify by default. I don't think it's possible for past data to change

Reply:
Resolved offline.

Reply:
resolved off-line